### PR TITLE
thunderbolt: add configuration option to force retimer enumeration

### PR DESCRIPTION
**Pull request by djcampello**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-device.h [hunk 1]
```diff
@@ -447,6 +447,15 @@ typedef guint64 FuDeviceInternalFlags;
...
 #define FU_DEVICE_INTERNAL_FLAG_MD_SET_SIGNED (1ull << 23)
 
+/**
+ * FU_DEVICE_INTERNAL_FLAG_FORCE_ENUMERATION:
+ *
+ * Forces composite device components to be enumerated.
+ *
+ * Since: 1.7.7
+ */
+#define FU_DEVICE_INTERNAL_FLAG_FORCE_ENUMERATION (1ull << 24)
```
Comment (hughsie) [inline libfwupdplugin/fu-device.h:457]: is this something we are going to use outside the tbt plugin?

Comment (djcampello) [reply libfwupdplugin/fu-device.h:457]: probably not

Comment (hughsie) [reply libfwupdplugin/fu-device.h:457]: fu_device_register_private_flag to the rescue

File: plugins/thunderbolt/fu-plugin-thunderbolt.c [hunk 2]
```diff
@@ -36,6 +37,8 @@ fu_plugin_thunderbolt_device_created(FuPlugin *plugin, FuDevice *dev, GError **e
 	fu_plugin_add_rule(plugin,
 			   FU_PLUGIN_RULE_INHIBITS_IDLE,
 			   "thunderbolt requires device wakeup");
+	if (fu_plugin_get_config_value_boolean(plugin, "RetimerOfflineMode"))
+		fu_device_set_private_flags(dev, FU_THUNDERBOLT_DEVICE_FLAG_FORCE_ENUMERATION);
```
Comment (hughsie) [inline plugins/thunderbolt/fu-plugin-thunderbolt.c:41]: fu_device_add_private_flag no?

Comment (djcampello) [reply plugins/thunderbolt/fu-plugin-thunderbolt.c]: ups overlooked.... sorry for the delay I had to setup my github access in another laptop

**Review by hughsie — APPROVED** [review]

**Pull request merged**